nifi-nar-bundles/nifi-single-user-iaa-providers-bundle/nifi-single-user-iaa-providers/src/main/java/org/apache/nifi/authorization/single/user/SingleUserAuthorizer.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.authorization.single.user;
+
+import org.apache.nifi.authorization.AuthorizationRequest;
+import org.apache.nifi.authorization.AuthorizationResult;
+import org.apache.nifi.authorization.Authorizer;
+import org.apache.nifi.authorization.AuthorizerConfigurationContext;
+import org.apache.nifi.authorization.AuthorizerInitializationContext;
+import org.apache.nifi.authorization.exception.AuthorizationAccessException;
+import org.apache.nifi.authorization.exception.AuthorizerCreationException;
+import org.apache.nifi.authorization.exception.AuthorizerDestructionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Single User Authorizer
+ */
+public class SingleUserAuthorizer implements Authorizer {
```

Comment:
Is there some way this could be linked together with the identity from the SingleUserLoginIdentityProvider?

This is really more of a no-op authorizer that just authorizes everything regardless of the user identity. For example, if a request was made with a client cert then it would be authorized to do anything, even if that identity was not the username of the SingleUserLoginIdentityProvider. Maybe not an issue if this is only intended for local use.

I'm not exactly sure how the coordination between identity providers and authorizers would work though, since they purposely don't know about each other. The only idea that comes to mind is if the single-user identity came from nifi.properties since they both have access to that. Alternatively, could maybe use the existing file-based providers, but have someway to setup the initial user / initial admin with the single-user identity, then eliminate the need for a new Authorizer implementation.

Reply:
Thanks for the feedback @bbende!  I considered some options for integration with the the existing file-based providers, but part of the challenge is having the generated username available to the file-based providers.

One alternative solution is to use a standard default username and set that value in the `Initial Admin Identity` property of the file-based provider.  That would remove the need for the username generation in the `SingleUserLoginIdentityProvider`, but it would also set a default username across the board.

On the other hand, this question could be addressed in future issues when it comes to changing the default configuration files.  With the goal here of providing single user access, a user would have to perform additional work to authenticate using client certificates, so I'm not sure if authorizing client certificate access should be a concern.

Reply:
Yea I agree that the client cert example is probably not realistic, but I guess my general concern was just providing an authorizer that allows everything. I assume the goal here is to eventually have this enabled by default, so I am picturing a user who is trying to switch away from the defaults and not knowing, or forgetting, and leaving this authorizer enabled.

I think the only way to make the username available across the board is if the single-user concept was in the framework at a higher level then the provider implementations, and by the time the provider's load their config, it was already populated with the single-user stuff.

Something like in nifi.properties...
```
nifi.security.single.user.mode.enabled=true
nifi.security.single.user.identity=
nifi.security.single.user.password=
```
The framework would generate these values and write them out to nifi.properties, then the identity provider and authorizer would get an instance of NiFiProperties and get these values and fail if they weren't populated. Maybe password would not be in nifi.properties and still in the identity provider (?).

I obviously haven't put that much thought into this, and not trying to derail your efforts, just throwing out ideas. If you don't think there is sufficient concern about the authorizer, I won't get too hung up on it.

Reply:
Thanks for the additional feedback, it is a challenge to find the right balance of security and ease of use for this implementation.  I would prefer to avoid introducing new properties into `nifi.properties` so that this can be remain modular.

The concern about authorizing all actions with the `SingleUserAuthorizer` is worth considering.  What about introducing a way to make it depend on the `SingleUserLoginIdentityProvider` being configured?  At a basic level, this could require having `nifi.security.user.login.identity.provider` set to `single-user-provider` and ensuring that the expected class is configured as a provider in `login-identity-providers.xml`?  Something along that line would ensure that the `SingleUserAuthorizer` only works with `SingleUserLoginIdentityProvider` and cannot be used with other authentication strategies.

Reply:
Yea that seems like a reasonable way to ensure the new authorizer is only used for the intended situation.

Reply:
Thanks @bbende, I will change `SingleUserAuthorizer` to check for `SingleUserLoginIdentityProvider`.

Reply:
@bbende I pushed an update so that `SingleUserAuthorizer` reads the configured `nifi.security.user.login.identity.provider` and parses `login-identity-providers.xml` to require that the provider class is `SingleUserLoginIdentityProvider`.  This should effectively prevent `SingleUserAuthorizer` from being used with any other login configuration.

Please let me know if you have any other comments or concerns.